OpenRA.Game/Player.cs

```diff
@@ -244,28 +249,40 @@ public bool IsAlliedWith(Player p)
 			return RelationshipWith(p) == PlayerRelationship.Ally;
 		}
 
-		public static Color PlayerRelationshipColor(Actor a)
+		/// <summary>Returns <see cref="color"/>, ignoring player relationship colors.</summary>
+		public static Color GetColor(Player p) => p.color;
+
+		public static void SetupRelationshipColors(Player[] players, Player observer)
 		{
-			var renderPlayer = a.World.RenderPlayer;
-			var player = renderPlayer ?? a.World.LocalPlayer;
-			if (player != null && !player.Spectating)
-			{
-				var effectiveOwner = a.EffectiveOwner;
-				var apparentOwner = a.Owner;
-				if (effectiveOwner != null && effectiveOwner.Disguised && !a.Owner.IsAlliedWith(renderPlayer))
-					apparentOwner = effectiveOwner.Owner;
+			foreach	(var p in players)
+				p.Color = PlayerRelationshipColor(p, observer);
+		}
 
-				if (apparentOwner == player)
-					return ChromeMetrics.Get<Color>("PlayerStanceColorSelf");
+		/// <summary>Returns <see cref="Color"/> of the <paramref name="actor"/>, taking <see cref="Actor.EffectiveOwner"/> into account.</summary>
+		public static Color ActorColor(Actor actor)
+		{
+			var effectiveOwner = actor.EffectiveOwner;
+			if (effectiveOwner != null && effectiveOwner.Disguised && actor.World.RenderPlayer != null)
+				return effectiveOwner.Owner.Color;
 
-				if (apparentOwner.IsAlliedWith(player))
-					return ChromeMetrics.Get<Color>("PlayerStanceColorAllies");
+			return actor.Owner.Color;
+		}
 
-				if (!apparentOwner.NonCombatant)
-					return ChromeMetrics.Get<Color>("PlayerStanceColorEnemies");
-			}
+		public static Color PlayerRelationshipColor(Player observee, Player observer)
```

Comment:
`observer` &rarr; `viewer` and `observee` &rarr; `player` for consistency with the selection/rendering code please.

Reply:
IMO there should also be a comment warning that this is a relatively slow method (`ChromeMetrics.Get` calls down into `FieldLoader`) so should not be used on a hot path.

Reply:
I can just make it non-public